http/transport.go

```diff
@@ -27,8 +42,100 @@ func NewDefaultTransport() *http.Transport {
 }
 
 // NewDefaultTransportWithTLS creates a new transport with the specified TLS configuration.
-func NewDefaultTransportWithTLS(tlsConfig *tls.Config) *http.Transport {
-	t := NewDefaultTransport()
+func NewDefaultTransportWithTLS(tlsConfig *tls.Config, dialer *net.Dialer) *http.Transport {
+	if dialer == nil {
+		dialer = &net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}
+	}
+	t := NewDefaultTransport(dialer)
 	t.TLSClientConfig = tlsConfig
 	return t
 }
+
+// Control is called after DNS lookup, but before the network connection is
+// initiated.
+func Control(urlValidator furl.Validator) func(network, address string, c syscall.RawConn) error {
+	return func(network, address string, c syscall.RawConn) error {
+		host, _, err := net.SplitHostPort(address)
+		if err != nil {
+			return err
+		}
+
+		ip := net.ParseIP(host)
+		return urlValidator.ValidateIP(ip)
+	}
+}
+
+// NewDefaultClientWithTLS creates a tls client with sane defaults.
+func NewDefaultClientWithTLS(tlsConfig *tls.Config, urlValidator furl.Validator) *http.Client {
+
+	// These defaults are copied from http.DefaultTransport.
+	return &http.Client{
+		Transport: NewDefaultTransport(&net.Dialer{
```

Comment:
Is this supposed to use the `tlsConfig` ... like with `NewDefaultTransportWithTLS`?